lib/appsignal.ex

```diff
@@ -53,12 +53,15 @@ defmodule Appsignal do
     # calling itself once it reached `Application.put_env` in
     # `Appsignal.Config`.
     spawn(fn ->
-      :ok = Appsignal.Nif.stop()
-      :ok = initialize()
+      Appsignal.Nif.stop()
+      initialize()
     end)
+
+    :ok
```

Comment:
Looks like this breaks the `release_upgrade_test` because it expects the PID to be returned and it no longer does that.

Reply:
I tried to make the test pass another way, by checking which new PID was alive after the config_change function was called, but at random times some tests register more than one new PID and it's difficult to determine which is the one we want to know about. Not sure how else to track this.

```patch
From edc67142d52d964ecda6df8c72ac58e9528b91b6 Mon Sep 17 00:00:00 2001
From: Tom de Bruijn <tom@tomdebruijn.com>
Date: Mon, 18 Jul 2022 11:46:22 +0200
Subject: [PATCH] Fix release_upgrade test after config_change

The config_change function no longer returns a PID. Because of that we
don't know the PID of the new spawned process to monitor.

Alternatively the config_change function can return a tuple of
`{:ok, pid}`, but I'm not sure if that would work. I'm unfamiliar with
the return value this function, but the docs it only returns `:ok`.

https://hexdocs.pm/elixir/1.13.4/Application.html#c:config_change/3
---
 test/appsignal/release_upgrade_test.exs | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git test/appsignal/release_upgrade_test.exs test/appsignal/release_upgrade_test.exs
index 0410e6d6..28e241a5 100644
--- test/appsignal/release_upgrade_test.exs
+++ test/appsignal/release_upgrade_test.exs
@@ -24,8 +24,20 @@ test "config_change/3" do
         |> Map.put(:name, "AppSignal test suite app v2")

       with_config(new_config, fn ->
+        pids_before = Process.list()
         # Hot reload / upgrade
-        config_reload_pid = Appsignal.config_change([], [], [])
+        Appsignal.config_change([], [], [])
+
+        pids_after = Process.list()
+        new_pids = pids_after -- pids_before
+
+        if length(new_pids) > 1 do
+          raise "More than one new process started (#{length(new_pids)}).
+            There should only be one new process, otherwise we can't tell
+            which one restarted the config."
+        end
+
+        config_reload_pid = List.first(new_pids)
         # The config is reloaded in a separate process so we wait for it here
         assert Process.alive?(config_reload_pid)

--
2.36.0
```

Reply:
I suppose it could be split in two: `do_config_change`, which returns `pid`, and `config_change`, a wrapper that calls `do_config_change`, ignores its return value, and returns `:ok`. Then we test `do_config_change`, and not its wrapper, the way we did before.